src/Data_Type_Adjuster.cxx

```diff
@@ -8,60 +8,60 @@
 
 // Note: the FITS workaround should be removed when we update to a
 // sufficiently recent version of CFITSIO.
-std::map<size_t, tablator::Data_Type>
-tablator::Data_Type_Adjuster::determine_datatypes_for_writing(
-        const Format &format) const {
+std::vector<tablator::Data_Type>
+tablator::Data_Type_Adjuster::get_datatypes_for_writing(
+        const Format::Enums &enum_format) const {
     auto columns = table_.columns;
-    std::map<size_t, tablator::Data_Type> alt_datatype_lookup;
-    bool format_ipac = (format.enum_format == Format::Enums::IPAC_TABLE);
-    bool format_votable = (format.enum_format == Format::Enums::VOTABLE);
-    bool adjust_uint64 = (format_ipac || format_votable ||
-                          (format.enum_format == Format::Enums::FITS));
+    std::vector<tablator::Data_Type> alt_datatypes;
+    bool format_ipac = (enum_format == Format::Enums::IPAC_TABLE);
+    bool format_votable = (enum_format == Format::Enums::VOTABLE);
+    bool format_fits = (enum_format == Format::Enums::FITS);
+    bool adjust_uint64 = (format_ipac || format_votable || format_fits);
 
-    // Skip the null bitfield flag
-    for (size_t col = 1; col <= columns.size(); ++col) {
-        if ((columns[col].type == tablator::Data_Type::UINT64_LE) && adjust_uint64) {
+    for (size_t col = 0; col <= columns.size(); ++col) {
```

Comment:
Wouldn't it be simpler to have the loop body retrieve the original type, adjust it to produce a new type, and only then append it to `alt_datatypes`, rather than appending first and then adjusting the `back()` of the vector?

Reply:
I think I did that because it ensures that columns which keep orig_datatype are handled properly; otherwise they risk falling through cracks between the ifs.  Unless I'm missing something.

Reply:
I guess I could check the size of the alt_datatypes vector....

Reply:
Just to be clear, I mean something like:
```c++
for (size_t col = 0; col <= columns.size(); ++col) {
    auto orig_datatype = columns[col].type;
    auto adjusted_datatype = orig_datatype;
    if (salad) {
        ...
        adjusted_datatype = tablator::Data_Type::CHAR;
    } else {
        ...
    }
    alt_datatypes.emplace_back(adjusted_datatype); 
}
```
I don't see any cracks. (Famous last words?)

Reply:
Ah, yep, I was missing that.